src/dao/MoodDao.java

```diff
@@ -45,6 +89,34 @@ public MoodDao() {
      * executing the query
      */
     public Mood getMoodById(int moodId) throws DaoException {
-        return null;
+        BufferedReader reader = null;
+        Mood rtnValue = null;
+        try {
+            String ln = "";
+            reader = new BufferedReader(new FileReader("data/Moods.csv"));
+            reader.readLine();
+
+            while ((ln = reader.readLine()) != null) {
```

Comment:
This is fine, but will keep iterating over the file even after the mood is found. This is not a big deal for our project, but if there are a lot records this approach will impede performance. A common way around this is to have a boolean that is set true only when the record is found and to test that boolean in the while loop. For example:

    boolean found = false;
    while ((ln = reader.readLine()) != null && !found) {
        // code
        if(md.getId() == moodId){ 
            rtnValue = md;
            found = true;
        }
    }